src/playdate/sound.nim

```diff
@@ -132,12 +134,38 @@ proc `offset=`*(this: FilePlayer, offset: float32) =
     privateAccess(PlaydateSoundFileplayer)
     playdate.sound.fileplayer.setOffset(this.resource, offset.cfloat)
 
+proc privateFilePlayerFinishCallback(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl, raises: [].} =
+    try:
+        let filePlayer = cast[FilePlayer](userdata)
+        if filePlayer.finishCallback != nil:
+            filePlayer.finishCallback(filePlayer)
+    except:
+        echo "Error while calling FilePlayer finish callback"
+
+proc setFinishCallback*(this: FilePlayer, callback: PDFilePlayerCallbackFunction) =
+    privateAccess(PlaydateSound)
+    privateAccess(PlaydateSoundFileplayer)
+    this.finishCallback = callback
+    if callback == nil:
+        playdate.sound.sampleplayer.setFinishCallback(this.resource, nil, nil)
```

Comment:
This is very suspicious: *sample*Player is used in a *file*player proc. If intended, it should be renamed. But probably this was not intended. Over on the bindings-side PlaydateSoundFileplayer.setFinishCallback is still commented